core/ArchiveProcessor/LoaderLock.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\ArchiveProcessor;
+
+use Piwik\Archive\ArchiveInvalidator;
+use Piwik\Cache;
+use Piwik\Common;
+use Piwik\Concurrency\Lock;
+use Piwik\Concurrency\LockBackend;
+use Piwik\Config;
+use Piwik\Container\StaticContainer;
+use Piwik\Context;
+use Piwik\DataAccess\ArchiveSelector;
+use Piwik\DataAccess\ArchiveTableCreator;
+use Piwik\DataAccess\ArchiveWriter;
+use Piwik\DataAccess\Model;
+use Piwik\DataAccess\RawLogDao;
+use Piwik\Date;
+use Piwik\Db;
+use Piwik\Option;
+use Piwik\Period;
+use Piwik\Piwik;
+use Piwik\SettingsServer;
+use Piwik\Site;
+use Psr\Log\LoggerInterface;
+use Piwik\CronArchive\SegmentArchiving;
+
+/**
+ * This class uses PluginsArchiver class to trigger data aggregation and create archives.
+ */
+class LoaderLock
+{
+
+    const MAX_LOCK_TIME = 60; //in seconds
+    protected $id;
+
+    public function __construct($id)
+    {
+        if (strlen($id) > 64) {
+            throw new \Exception('exceeds length');
```

Comment:
can we here instead change the ID to fit into 64 characters? Like `substr($id,0,32).md5($id)`? This will make it more developer friendly and avoid if we accidentally create a too long id